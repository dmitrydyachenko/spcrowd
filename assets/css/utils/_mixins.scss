@import 'mixins/breakpoints';
@import 'mixins/center';
@import 'mixins/clearfix';
