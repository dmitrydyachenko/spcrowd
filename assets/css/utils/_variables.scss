/**
 * Breakpoints
 */

$breakpoints: (
	x-small: 479px, 
    small: 639px,
    medium: 780px,
	large: 1023px,
    x-large: 1365px,
    xx-large: 1919px
);

/**
 * Tweakpoints
 */

$tweakpoints: ();

/**
 * Other variables
 */
$baseline: 24px;

/**
 * Colours
 */
$white: #FFF;
$black: #000;
$grey: #666;

/**
 * Paths to Sharepoint assets 
 */
 $fontspath: "../fonts";
 $imgpath: "../img";
