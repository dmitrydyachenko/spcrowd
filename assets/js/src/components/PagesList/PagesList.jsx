/* External libraries */
import $ from 'jquery';
import React from 'react';

/* CSS styles */
import Styles from './PagesList.scss';

class PagesList extends React.Component {
	static propTypes = {
		data: React.PropTypes.arrayOf(React.PropTypes.object).isRequired,
		fontsize: React.PropTypes.string,
		guid: React.PropTypes.string
	};

	constructor(props) {
		super(props);

		this.state = {
			data: this.props.data
		};
	}

	componentWillReceiveProps(nextProps) {        
		this.setState({ data: nextProps.data });
	}

	render() {
		const self = this;
		const pages = self.state.data;
		const mainContent = pages && pages.length > 0 ? pages.map((item, i) => 
			<div key={i} className={Styles.item} style={{ fontSize: self.props.fontsize }}>
				{`${i + 1}. ${item.Title}`}
			</div>
		) : null;

		return (
			<div className={Styles.container}>
				<p className={Styles.header}>
					{self.props.guid}
				</p>
				<p className={Styles.header}>
					{'Site pages:'}
				</p>
				<div className={Styles.content}>
					{mainContent}
				</div>
			</div>
		);
	}
}

export default PagesList;
