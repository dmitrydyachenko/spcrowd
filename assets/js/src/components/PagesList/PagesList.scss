.container {
    display: block;

    .header {
        display: block;
        font-size: 14px;
        font-weight: 400;
        -webkit-font-smoothing: antialiased;
        color: #333333;
    }

    .content {
        display: block;

        .item {
            display: block;
            font-size: 14px;
            font-weight: 400;
            -webkit-font-smoothing: antialiased;
            color: #333333;
        }
    }
}