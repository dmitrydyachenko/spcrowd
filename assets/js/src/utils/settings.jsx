export const IMGPATH = 'img';
export const PAGESLIST = 'Pages';
export const STYLELIBRARY = 'Style Library';
export const DOCUMENTSLIBRARY = 'Documents';
export const SETTINGSLIST = 'Settings';
export const PUBLISHINGIMAGESLIBRARY = 'PublishingImages';
export const NAMESPACES = {
	group: 'Golin',
	prefix: 'Golin'
};
export const XMLFILENAMES = {
	fields: 'Fields.xml',
	lists: 'Lists.xml',
	contentTypes: 'ContentTypes.xml',
	groups: 'Groups.xml'
};
export const EXCELSHEETNAMES = {
	fields: 'Fields',
	lists: 'Lists',
	contentTypes: 'ContentTypes',
	groups: 'Groups',
	fieldsMapping: 'FieldsMapping'
};