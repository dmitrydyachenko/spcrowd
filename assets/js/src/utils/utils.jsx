import Moment from 'moment';

export function GetAssetsPath(projectName) {
	return `${_spPageContextInfo.siteAbsoluteUrl}/Style%20Library/${projectName}/`;
}

export function GetEqNeqOrAndFilter(fields, values, comp, cond) {
	let filterString = '';

	for (let i = 0; i < fields.length; i++) {
		for (let j = 0; j < values.length; j++) {
			filterString += (`${fields[i]} ${comp || 'eq'} ${values[j] === 1 || values[j] === 0 ? values[j] : `'${values[j]}'`} ${cond || 'or'} `);
		}
	}

	return filterString.trim().replace(/ or$| and$/, '');
}

export function GetSubStringOfFilter(value, fields, cond) {
	let filterString = '';

	for (let i = 0; i < fields.length; i++) {
		const condition = cond || 'or';
		filterString += (`substringof('${value}', ${fields[i]}) ${condition} `);
	}

	return filterString.trim().replace(/ or$| and$/, '');
}

export function MergeObjects(obj1, obj2) {
	const obj3 = {};

	if (obj1) {
		Object.keys(obj1).forEach((key) => {
			obj3[key] = obj1[key];
		});
	}

	if (obj2) {
		Object.keys(obj2).forEach((key) => {
			obj3[key] = obj2[key];
		});
	}

	return obj3;
}

export function JoinQuery(query1, query2, cond) {
	return (query1 && query2 ? `(${query1}) ${cond} (${query2})` : (query1 || query2));
}

export function CompareByTitle(a, b) {
	return (a.title < b.title ? -1 : (a.title > b.title ? 1 : 0));
}

export function CompareByCreated(a, b) {
	const newA = Moment(a.Created);
	const newB = Moment(b.Created);
	const diff = newA.diff(newB);
	return diff < 0 ? 1 : diff > 0 ? -1 : 0;
}

export function CreateMarkup(htmlString) {
	return {
		__html: htmlString
	};
}

export function RedirectToPage(pageUrl) {
	window.location.href = pageUrl;
}

export function ListUserProfileProperties(results) {
	for (let i = 0; i < results.length; i++) {
		console.log(`Key: ${results[i].Key} Value: ${results[i].Value}`);
	}
}

export function DetectIE() {
	const ua = window.navigator.userAgent;
	const msie = ua.indexOf('MSIE ');

	if (msie > 0) {
		return parseInt(ua.substring(msie + 5, ua.indexOf('.', msie)), 10);
	}

	const trident = ua.indexOf('Trident/');

	if (trident > 0) {
		const rv = ua.indexOf('rv:');
		return parseInt(ua.substring(rv + 3, ua.indexOf('.', rv)), 10);
	}

	const edge = ua.indexOf('Edge/');

	if (edge > 0) {
		return parseInt(ua.substring(edge + 5, ua.indexOf('.', edge)), 10);
	}

	return false;
}

export function GetFileNameFromUrl(url) {
	const splittedUrl = url.split('/');
	const splittedUrlLength = splittedUrl.length;
	return splittedUrlLength > 0 ? splittedUrl[splittedUrlLength - 1] : '';
}

export function IfArrayContainsObject(array, property, value) {
	const index = -1;

	if (value) {
		for (let i = 0; i < array.length; i++) {
			if (array[i][property] === value) {
				return i;
			}
		}
	}

	return index;
}

function s4() {
	return Math.floor((1 + Math.random()) * 0x10000).toString(16).substring(1);
}

export function GenerateGuid() {
	return `${s4()}${s4()}-${s4()}-${s4()}-${s4()}-${s4()}${s4()}${s4()}`;
}

export function EqualObjects(obj1, obj2, field) {
	return obj1 && obj2 ? JSON.stringify(obj1[field]) === JSON.stringify(obj2[field]) : (!obj1 && !obj2);
}

export function SelectSrcValue(imageString) {
	if (imageString) {
		const imgArray = imageString.split(' ');

		for (let i = 0; i < imgArray.length; i++) {
			if (imgArray[i].startsWith('src')) {
				const srcArray = imgArray[i].split('"');
				return srcArray.length > 0 ? srcArray[1] : '';    
			}
		} 
	}

	return '';
}

export function ArrayAdiff(a1, a2) {
	return a1.filter(x => a2.indexOf(x) < 0);
}