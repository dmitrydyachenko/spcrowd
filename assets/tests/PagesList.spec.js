// Link.react-test.js
import React from 'react';
import Renderer from 'react-test-renderer';
import PagesList from '../js/src/components/PagesList/PagesList';
import { GenerateGuid } from '../js/src/utils/utils';
import { PAGESLIST } from '../js/src/utils/settings';

test('PagesList', () => {
	const data = [
		{ Title: "Page 1" },
		{ Title: "Page 2" }
	];

	const component = Renderer.create(
		<PagesList data={data} guid={GenerateGuid()} fontsize={"64"} />
	);

	expect(component.toJSON()).toMatchSnapshot();
});
