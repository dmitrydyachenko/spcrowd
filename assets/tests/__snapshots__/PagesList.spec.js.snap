exports[`test PagesList 1`] = `
<div
  className="container">
  <p
    className="header">
    89d855b6-d1d7-6e35-51bb-1b61656af352
  </p>
  <p
    className="header">
    Site pages:
  </p>
  <div
    className="content">
    <div
      className="item"
      style={
        Object {
          "fontSize": "64",
        }
      }>
      1. Page 1
    </div>
    <div
      className="item"
      style={
        Object {
          "fontSize": "64",
        }
      }>
      2. Page 2
    </div>
  </div>
</div>
`;
